packages/http-server-csharp/src/lib/utils.ts

```diff
@@ -1441,27 +1465,66 @@ export class CSharpOperationHelpers {
           const { defaultValue: itemDefault } = this.getTypeInfo(program, value);
           defaults.push(itemDefault);
         }
-        return {
-          typeReference: code`${csharpType.getTypeReference(myEmitter.getContext().scope)}[]`,
-          defaultValue: `[${defaults.join(", ")}]`,
-          nullableType: csharpType.isNullable,
-        };
+        const collectionType = CSharpServiceOptions.getInstance().collectionType;
+
+        switch (collectionType) {
+          case CollectionType.IEnumerable:
+            return {
+              typeReference: code`IEnumerable<${csharpType.getTypeReference()}>`,
```

Comment:
All of these need to use `myEmitter.getContext()?.scope` when getting the type reference.  Also, though the result may be cached, we should probably calcualate this once and use it in typeReference and defaultValue.